shared/profile/edit-avatar/index.desktop.js

```diff
@@ -1,33 +1,388 @@
+  }
+
+  _onRangeChange = (e: SyntheticInputEvent<any>) => {
+    const scale = e.currentTarget.value
+    const scaledImageHeight = Math.round(this.state.originalImageHeight * scale)
+    const scaledImageWidth = Math.round(this.state.originalImageWidth * scale)
+
+    this.setState({
+      scale,
+      scaledImageHeight,
+      scaledImageWidth,
+    })
+  }
+
+  _onMouseDown = (e: SyntheticMouseEvent<any>) => {
+    this.setState({
+      dragStartX: e.pageX,
+      dragStartY: e.pageY,
+      dragStopX:
+        this._image && this._image.style.left ? parseInt(this._image.style.left, 10) : this.state.dragStopX,
+      dragStopY:
+        this._image && this._image.style.top ? parseInt(this._image.style.top, 10) : this.state.dragStopY,
+      dragging: true,
+      offsetLeft: this._image ? this._image.offsetLeft : this.state.offsetLeft,
+      offsetTop: this._image ? this._image.offsetTop : this.state.offsetTop,
+    })
+  }
+
+  _onMouseUp = () => {
+    this.setState({
+      dragStopX:
+        this._image && this._image.style.left ? parseInt(this._image.style.left, 10) : this.state.dragStopX,
+      dragStopY:
+        this._image && this._image.style.top ? parseInt(this._image.style.top, 10) : this.state.dragStopY,
+      dragging: false,
+      offsetLeft: this._image ? this._image.offsetLeft : this.state.offsetLeft,
+      offsetTop: this._image ? this._image.offsetTop : this.state.offsetTop,
+    })
+  }
+
+  _onMouseMove = (e: SyntheticMouseEvent<any>) => {
+    if (!this.state.dragging || this.state.submitting) return
+
+    const dragLeft = this.state.dragStopX + e.pageX - this.state.dragStartX
+    const dragTop = this.state.dragStopY + e.pageY - this.state.dragStartY
+    const dragLeftLimit = AVATAR_SIZE - AVATAR_BORDER_WIDTH * 2 - this.state.scaledImageWidth
+    const dragTopLimit = AVATAR_SIZE - AVATAR_BORDER_WIDTH * 2 - this.state.scaledImageHeight
+
+    this.setState({
+      offsetLeft: clamp(dragLeft, dragLeftLimit, 0),
+      offsetTop: clamp(dragTop, dragTopLimit, 0),
+    })
+  }
+
+  _onSave = () => {
+    this.setState({submitting: true})
+
+    const x = this.state.offsetLeft * -1
+    const y = this.state.offsetTop * -1
+    const rH =
+      this.state.scaledImageHeight !== 0 ? this.state.originalImageHeight / this.state.scaledImageHeight : 1
+    const rW =
+      this.state.scaledImageWidth !== 0 ? this.state.originalImageWidth / this.state.scaledImageWidth : 1
+    const crop = {
+      x0: Math.round(x * rW),
+      x1: Math.round((x + AVATAR_SIZE - AVATAR_BORDER_WIDTH * 2) * rW),
+      y0: Math.round(y * rH),
+      y1: Math.round((y + AVATAR_SIZE - AVATAR_BORDER_WIDTH * 2) * rH),
+    }
+    this.props.onSave(this.state.imageSource, crop)
+  }
+
+  render() {
+    return (
+      <MaybePopup
+        onClose={this.props.onClose}
+        styleCover={collapseStyles([
+          styles.cover,
+          {
+            cursor: !this.state.hasPreview ? 'default' : this.state.dragging ? 'grabbing' : 'grab',
```

Comment:
@chrisnojima, this doesn't work, but I don't understand why. Can you explain why it doesn't?